### ISSUE TITLE
Why use the logging class?

### ISSUE THREAD

**Issue by gigo6000** [pinfirestudios/yii2-bugsnag#11]
If the errors are sent to Bugsnap through the errorHandler, I don't understand why the logging part is needed.

**Comment by jcherniak** [comment]
Can you elaborate on what you’re talking about?  We include an implementation of a Yii-logger so that we can send the full Yii logs to Bugsnag when an exception occurs.  This allows you to capture very helpful information when a crash happens.

Are you talking about this or something else?

**Comment by gigo6000** [comment]
To give you some context after implementing the extension in a project we got tons of errors reported to Bugsnag. Yii2 basically convert all errors into exceptions but we don't really want to see 404s reported for example, so I customized the `BugsnagComponent` to filter those exceptions.

I think is not very clear to me when errors are reported to Bugsnag from the logs or directly, so that's why I ask why we need to report from the logs. Looking at the code I see there's a flag called `$inExceptionHandler` to prevent double reporting, and on the base `ErrorHandler` class, I see all the exceptions are logged so I'm confused.

**Comment by kesselb** [comment]
@gigo6000 Thank you for asking that question. I was running into the same today. We have many 404 exceptions in bugsnag. I tried to filter these exceptions like described here: https://www.yiiframework.com/doc/guide/2.0/en/runtime-logging#message-filtering but it didn't work. 

After some debugging i found out that `\pinfirestudios\yii2bugsnag\BugsnagWebErrorHandler` change the exception category to Bugsnag notified exception. Default yii error handler with the bugsnag log target works fine.

**Comment by jcherniak** [comment]
@kesselb, you are correct, that will simply log exceptions.  If you want full context including stacktrace info and other logged variables, you need the custom error handler to catch the actual exception, not the log message.

**Comment by jcherniak** [comment]
The logging portion allows you to get Trace level logs as a custom tab when exceptions are reported, allowing you to step backwards.  You need both pieces to get full stack traces and trace level logging.

**Comment by kesselb** [comment]
`\pinfirestudios\yii2bugsnag\BugsnagWebErrorHandler` will break some of the default yii2 logging functionality (e.g. message filtering). I would not recommend it if you plan to use other log targets (like file or db).
